File: src/agents/claude-md.ts
import { writeFileSync } from "fs";
import { join } from "path";
import type { AgentName } from "../model/agent.js";
import { PERSONALITIES } from "../model/personality.js";

export function writeClaudeMd(worktreePath: string, agentName: string): void {
  const personality = PERSONALITIES[agentName.toLowerCase() as AgentName];
  const personalitySection = personality
    ? `\n## Personality\n${personality.systemPrompt}\n`
    : "";

  const content = `# fm pipeline

## Project Overview
A terminal dashboard CLI (\`fm\`) that aggregates work items from Trello, Linear, and Jira.
Built with TypeScript, React Ink (terminal UI), and Node.js.

## Tech Stack
- **Runtime**: Node.js (ES2022)
- **Language**: TypeScript (strict mode)
- **UI**: React 18 + Ink 5 (terminal rendering)
- **Build**: tsup
- **Test**: vitest
- **Package manager**: npm

## Conventions
- ESM modules (\`"type": "module"\` in package.json)
- Imports use \`.js\` extension (e.g., \`import { foo } from "./bar.js"\`)
- No default exports — use named exports
- Zod for runtime validation at system boundaries
- Models in \`src/model/\`, providers in \`src/providers/\`, UI in \`src/ui/\`

## Testing
- Test files: \`src/__tests__/*.test.ts\` or \`src/__tests__/*.test.tsx\`
- Use \`describe/it/expect\` from vitest
- Mock external APIs, use real file system with temp directories
- Run: \`npm test\`

## Commit Format
- Short imperative subject line (e.g., "Add login validation")
- Reference the work item ID in the commit body

## Agent Identity
You are **${agentName}**, an autonomous agent working in a git worktree.
Your changes will be submitted as a pull request for review.
${personalitySection}`;

  writeFileSync(join(worktreePath, "CLAUDE.md"), content);
}

File: src/agents/claude_md.rs
use anyhow::Result;
use std::path::Path;

use crate::model::agent::AgentName;
use crate::model::personality::personality;

pub fn write_claude_md(worktree_path: &Path, agent_name: AgentName) -> Result<()> {
    let p = personality(agent_name);
    let traits = p.traits.join(", ");

    let content = format!(
        r#"# work pipeline

## Project Overview
A terminal dashboard CLI (`work`) that aggregates work items from Trello, Linear, and Jira.
Built with Rust, Ratatui (terminal UI).

## Tech Stack
- **Language**: Rust
- **UI**: Ratatui + Crossterm
- **Build**: cargo
- **Test**: cargo test

## Conventions
- Modules in `src/model/`, providers in `src/providers/`, UI in `src/ui/`
- Use `anyhow` for error handling
- Use `serde` for serialization

## Testing
- Run: `cargo test`

## Commit Format
- Short imperative subject line (e.g., "Add login validation")
- Reference the work item ID in the commit body

## Agent Identity
You are **{display}**, an autonomous agent working in a git worktree.
Your changes will be pushed directly to main.

### Personality: {tagline}
- **Focus**: {focus}
- **Traits**: {traits}
- **Working style**: {system_prompt}
"#,
        display = agent_name.display_name(),
        tagline = p.tagline,
        focus = p.focus,
        traits = traits,
        system_prompt = p.system_prompt,
    );

    std::fs::write(worktree_path.join("CLAUDE.md"), content)?;
    Ok(())
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn claude_md_includes_focus_for_all_agents() {
        let dir = tempfile::tempdir().unwrap();
        for name in AgentName::ALL {
            write_claude_md(dir.path(), name).unwrap();
            let content = std::fs::read_to_string(dir.path().join("CLAUDE.md")).unwrap();
            let p = personality(name);
            assert!(
                content.contains("**Focus**:"),
                "{name} CLAUDE.md missing Focus field"
            );
            assert!(
                content.contains(p.tagline),
                "{name} CLAUDE.md missing tagline"
            );
            assert!(
                content.contains(name.display_name()),
                "{name} CLAUDE.md missing display name"
            );
        }
    }
}
